msvc/sdk-build.props

```diff
@@ -29,6 +29,7 @@
     <ApplicationTypeRevision Condition="'$(ApplicationTypeRevision)'==''">10.0</ApplicationTypeRevision>
     <UseStarboardSourceSdk Condition="'$(UseStarboardSourceSdk)'==''">true</UseStarboardSourceSdk>
     <IslandwoodDRT Condition="'$(IslandwoodDRT)'==''">false</IslandwoodDRT>
+    <StarboardLinkWholeArchive Condition="'$(StarboardLinkWholeArchive)'==''">true</StarboardLinkWholeArchive>
```

Comment:
>StarboardLinkWholeArchive [](start = 5, length = 25)

Is this something we would expect our 3rd party consumers to want by default? I imagine any of them that want categories will be sad if this is off (and they make a lib + dll I guess).... just wondering if we want this to be opt out instead of in.